File: Backend/src/main/java/com/ntg/backend/dto/requestDto/RequestDto.java
package com.ntg.backend.dto.requestDto;


import com.ntg.backend.entity.Request;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
public class RequestDto {
//    private Long requestId;
    private Long requestId;
    private LocalDate date;
    private Request.RequestStatus status;
    private String reason;
    private Long needyId;    // Add needyId
    private Long itemId;     // Add itemId


}

File: Frontend/src/app/components/interfaces/request-dto.ts
export interface RequestDto {
  requestId: number;
  date: string; // Use string for LocalDate (ISO 8601 format, e.g., "2024-11-22")
  status: RequestStatus;
  reason: string;
  needyId: number;
  itemId: number;
}

export enum RequestStatus {
  مرفوض = "مرفوض",
  قيد_المراجعه = "قيد_المراجعه",
  تم_الموافقه = "تم_الموافقه"
}
